hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go

```diff
@@ -1304,7 +1312,15 @@ func reconcileHostedControlPlane(hcp *hyperv1.HostedControlPlane, hcluster *hype
 	// always reconcile the release image (facilitates rolling forward)
 	hcp.Spec.ReleaseImage = hcluster.Spec.Release.Image
 
-	hcp.Spec.Configuration = hcluster.Spec.Configuration.DeepCopy()
+	if hcluster.Spec.Configuration != nil {
+		hcp.Spec.Configuration = hcluster.Spec.Configuration.DeepCopy()
```

Comment:
Why are we setting this to a deepcopy of itself?

Reply:
the destination is the configuration in the HostedControlPlane, the source is the configuration in the HostedCluster